diff --git a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
index bcf9b00..0000000 100644
--- a/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
+++ b/node_modules/expo-modules-core/android/ExpoModulesCorePlugin.gradle
@@ -69,6 +69,9 @@ ext.useExpoPublishing = {
   if (!project.plugins.hasPlugin('maven-publish')) {
     apply plugin: 'maven-publish'
   }
+  if (components.findByName("release") == null) {
+    return
+  }

   project.android {
